src/cpp/transport/tcp/RTCPMessageManager.cpp

```diff
@@ -491,56 +511,56 @@ void RTCPMessageManager::processCheckLogicalPortsRequest(
         SerializedPayload_t payload(static_cast<uint32_t>(
             CheckLogicalPortsResponse_t::getBufferCdrSerializedSize(response)));
         response.serialize(&payload);
-        sendData(p_channel_resource, CHECK_LOGICAL_PORT_RESPONSE, transaction_id, &payload, RETCODE_OK);
+        sendData(channel, CHECK_LOGICAL_PORT_RESPONSE, transaction_id, &payload, RETCODE_OK);
     }
 }
 
 ResponseCode RTCPMessageManager::processKeepAliveRequest(
-        TCPChannelResource *p_channel_resource,
+        std::shared_ptr<TCPChannelResource>& channel,
         const KeepAliveRequest_t &request,
         const TCPTransactionId &transaction_id)
 {
-    if (!p_channel_resource->connection_established())
+    if (!channel->connection_established())
     {
-        sendData(p_channel_resource, KEEP_ALIVE_RESPONSE, transaction_id, nullptr, RETCODE_SERVER_ERROR);
+        sendData(channel, KEEP_ALIVE_RESPONSE, transaction_id, nullptr, RETCODE_SERVER_ERROR);
     }
-    else if (IPLocator::getLogicalPort(p_channel_resource->locator()) == IPLocator::getLogicalPort(request.locator()))
+    else if (IPLocator::getLogicalPort(channel->locator()) == IPLocator::getLogicalPort(request.locator()))
     {
-        sendData(p_channel_resource, KEEP_ALIVE_RESPONSE, transaction_id, nullptr, RETCODE_OK);
+        sendData(channel, KEEP_ALIVE_RESPONSE, transaction_id, nullptr, RETCODE_OK);
     }
     else
     {
-        sendData(p_channel_resource, KEEP_ALIVE_RESPONSE, transaction_id, nullptr, RETCODE_UNKNOWN_LOCATOR);
+        sendData(channel, KEEP_ALIVE_RESPONSE, transaction_id, nullptr, RETCODE_UNKNOWN_LOCATOR);
         return RETCODE_UNKNOWN_LOCATOR;
     }
     return RETCODE_OK;
 }
 
 void RTCPMessageManager::processLogicalPortIsClosedRequest(
-        TCPChannelResource* p_channel_resource,
+        std::shared_ptr<TCPChannelResource>& channel,
```

Comment:
Using a reference instead of a copy can cause problems as the channel being freed while we are still using it.